File: 2025-and-before/0523-continuous-subarray-sum/0523-continuous-subarray-sum.java
class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
            
        HashMap<Integer, Integer> map = new HashMap<>();
        int pSum = 0, ans = 0, mod;
        map.put(0, -1);
        for (int i = 0; i < nums.length; i++) {
            pSum += nums[i];
            mod = (pSum % k + k)%k;
            
            if (map.containsKey(mod)) {
                ans = Math.max(ans, i - map.getOrDefault(mod,i));
            } else {
                map.put(mod, i);
            }
        }
        return ans >= 2;
    }
}

File: 2025-and-before/0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
class Solution
{
    public:

        bool checkSubarraySum(vector<int> &arr, int k)
        {
            unordered_map<int, int> mp;
            int p_sum = 0, mod;
            mp[0] = -1;
            for (int i = 0; i < arr.size(); i++)
            {
                p_sum += arr[i];

                mod = p_sum % k;

                if (mp.count(mod) == 1)
                {
                    if (i - mp[mod] >= 2)
                        return true;
                }
                
                else
                    mp[mod] = i;
            }
            return false;
        }
};